server/src/api.js

```diff
@@ -62,6 +62,19 @@ class Api {
     return {};
   }
 
+  /**
+   * @returns {FileInfo}
+   */
+  static deletePreview() {
+    const file = new FileInfo(`${Config.previewDirectory}preview.tif`);
+    const parent = new FileInfo(file.path);
```

Comment:
Since the deletion path is not coming from the browser, there's no need to check if it's outside the data directory. You can just keep lines 69 and 75.

Also, there's now duplicate code inside `Api.context`. It also deletes the preview image - just call this new method from there.